codegen/tests/simple/types.go

```diff
@@ -200,3 +229,39 @@ type Str string
 // Empty ...
 type Empty struct {
 }
+
+// PetA can be one of following types in runtime:
+// Cat
+// Dog
+type PetA interface {
+	// IsPetA returns true if it is instance of PetA
+	IsPetA() bool
```

Comment:
Method should be `isPetA()`:
1. We don't want to export the method we just want to mark the implementations.
1. We don't need to return anything; void will suffice.